app/src/main/java/com/example/itunesapp/viewModel/ItunesViewModel.kt

```diff
@@ -1,46 +1,27 @@
 package com.example.itunesapp.viewModel
 
-import android.app.Application
-import android.content.Context
-import android.util.Log
-import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MutableLiveData
 import androidx.lifecycle.ViewModel
-import com.example.itunesapp.database.DatabaseImpl
+import androidx.lifecycle.viewModelScope
 import com.example.itunesapp.model.ResultDatabase
-import com.example.itunesapp.network.ApiInstance
-import com.example.itunesapp.repositoryMapping.ResultMappingImpl
-import kotlinx.coroutines.GlobalScope
+import com.example.itunesapp.repository.Repository
+import com.example.itunesapp.repository.RepositoryInter
 import kotlinx.coroutines.launch
 
-class ItunesViewModel(application: Application) : AndroidViewModel(application) {
+class ItunesViewModel (private val repository: Repository): ViewModel() {
 
+    lateinit var searchSongsList:MutableLiveData<ArrayList<ResultDatabase>>
 
-    val searchItunesApi = ApiInstance.searchItunesApi
 
-    var databaseImpl:DatabaseImpl = DatabaseImpl.getDatabase(application)
-
-    val resultMappingImpl = ResultMappingImpl()
-
-
-    fun searchItem(){
-
-        GlobalScope.launch {
-            val result = searchItunesApi.searchSongs("#search")
-            if (result != null){
-                var searchItem = result.body()
-                if (searchItem != null) {
-
-                    var searchList:List<ResultDatabase> = resultMappingImpl.searchList(searchItem.results)
-                    Log.d("View Model@@@@@@@@@@",searchList.toString())
-                    databaseImpl.databaseDao().addSearchItems(searchList)
-                }
-
-
-            }
+    suspend fun searchItem(){
+        viewModelScope.launch {
+            repository.searchItem("#search")
         }
```

Comment:
Since we are launching a new coroutine, the `searchItem` function does not need to be a suspend function.